src/ripple/protocol/impl/SecretKey.cpp

```diff
@@ -199,6 +210,8 @@ generateSecretKey (KeyType type, Seed const& seed)
 
     if (type == KeyType::secp256k1)
     {
+        // FIXME: Avoid copying the seed into a uint128 key only to have
+        //        generateRootDeterministicPrivateKey copy out of it.
```

Comment:
I probably don't understand all the nuances of the data - why don't we need to `secure_erase` the `upk` below?

Reply:
We'd need to massage things a bit to do it, but it's possible. Perhaps in the interest of thoroughness we should, even if it's a bit of extra work.

As I've marked elsewhere in the file, ideally we should change `generateRootDeterministicPrivateKey` to just take a `Slice` and not a `uint128`, and have it return a `SecretKey` directly.